python/packages/nisar/static/rtc_anf_layers.py

```diff
@@ -0,0 +1,231 @@
+        routine used in geo2rdr (Newton-Raphson implementation). The following keys are
+        supported:
+
+        'threshold':
+          Absolute azimuth time convergence tolerance, in seconds. Defaults to 1e-8.
+
+        'maxiter':
+          Maximum number of Newton-Raphson iterations. Defaults to 50.
+    output_mode : isce3.geocode.GeocodeOutputMode or str, optional
+        Algorithm used for geocoding. Defaults to the area projection algorithm.
+    interp_method : isce3.core.DataInterpMethod or str, optional
+        Interpolation method used for geocoding. Only relevant when `output_mode` is set
+        to interpolation. Ignored otherwise. Defaults to biquintic interpolation.
+    memory_mode : isce3.core.GeocodeMemoryMode, optional
+        Block processing mode to use for geocoding. Defaults to ???.
+    min_block_size : int, optional
+        Minimum block size for geocoding, per thread, in bytes. The block size is chosen
+        dynamically based on the image dimensions to partition work uniformly across
+        available threads, bounded by `min_block_size` and `max_block_size`. The number
+        of threads may be controlled by the `OMP_NUM_THREADS` environment variable. Must
+        be <= `max_block_size`. Defaults to 33,554,432 (32 MiB).
+    max_block_size : int, optional
+        Maximum block size for geocoding, per thread, in bytes. The block size is chosen
+        dynamically based on the image dimensions to partition work uniformly across
+        available threads, bounded by `min_block_size` and `max_block_size`. The number
+        of threads may be controlled by the `OMP_NUM_THREADS` environment variable. Must
+        be >= `min_block_size`. Defaults to 268,435,456 (256 MB).
+    min_area_factor : float, optional
+        Minimum pixel area threshold for RTC, in dB. If set to NaN, no minimum is
+        enforced, which may lead to division by very small values and result in infinite
+        or inaccurate outputs. Defaults to NaN.
+    dem_upsample_factor : float, optional
+        DEM upsampling factor used for RTC. Defaults to 2.
+    algorithm : isce3.geometry.RtcAlgorithm or str, optional
+        Algorithm used for RTC. Defaults to the area projection algorithm.
+    area_beta_mode : isce3.geometry.RtcAreaBetaMode or str or None, optional
+        Method for estimating the area of the beta naught reference surface. Defaults to
+        ???.
+
+    Returns
+    -------
+    gamma0_to_beta0_factor : isce3.io.Raster
+        RTC scaling factor to normalize backscatter coefficients from gamma0 to sigma0,
+        accounting for local terrain.
+    gamma0_to_sigma0_factor : isce3.io.Raster
+        RTC scaling factor to normalize backscatter coefficients from gamma0 to beta0,
+        accounting for local terrain.
+    """
+    if geo2rdr_params is None:
+        geo2rdr_params = {}
+
+    geocode = isce3.geocode.GeocodeCFloat32()
```

Comment:
This will run faster if we geocode real-valued data instead of complex-valued.

Note that we don't care about the geocoded covariance output. The gamma0-to-beta0 and gamma0-to-sigma0 layers that we do care about should be unaffected.

```suggestion
    geocode = isce3.geocode.GeocodeFloat32()
```